adaptive/learner/averagelearner1D.py

```diff
@@ -0,0 +1,399 @@
+from copy import deepcopy
+from math import hypot
+
+import numpy as np
+import scipy.stats
+from sortedcollections import ItemSortedDict
+from sortedcontainers import SortedDict
+
+from adaptive.learner.learner1D import Learner1D, _get_intervals
+from adaptive.notebook_integration import ensure_holoviews
+
+
+class AverageLearner1D(Learner1D):
+    """Learns and predicts a noisy function 'f:ℝ → ℝ^N'.
+
+    New parameters (wrt Learner1D)
+    ------------------------------
+    delta : float
+        This parameter controls the resampling condition. A point is resampled
+        if its uncertainty is larger than delta times the smallest neighboring
+        interval.
+        We strongly recommend 0 < delta <= 1.
```

Comment:
Isn't this even required?

Reply:
Not really. If delta<0, the algorithm will only resample existing points and stop sampling new ones. If delta>1, the algorithm will mostly sample new points (larger delta means less resampling). These two regimes are not commonly desired, but one may use them in some very specific cases. That is why I decided to allow for any value of delta, although in general we recommend 0<delta<1 (more specifically, delta~0.2).